### PR TITLE
Add GatewayRoute CRD API types for v1beta2

### PR DESCRIPTION
**Pull request by karanvasnani**
*Issue #, if available:*
[aws-app-mesh-roadmap #37](https://github.com/aws/aws-app-mesh-roadmap/issues/37#issuecomment-614127518)

*Description of changes:*
Adding new `GatewayRoute` CRD API version for v1beta2(based on designs #150)

Note for reviewer:
The PR is separated into three commits, only the 2nd one needs to be reviewed while the other two are auto-generated.

- "auto-generate GatewayRoute CRD API" is auto-generated by
`kubebuilder create api --group appmesh --version v1beta2 --kind GatewayRoute`
- "Add Fields into GatewayRoute API" contains real manual changes
- "auto-generate yaml for GatewayRoute API" is auto-generated by `make manifests` and `make generate`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/appmesh/v1beta2/gatewayroute_types.go [hunk 1]
```diff
@@ -17,25 +17,126 @@ limitations under the License.
...
+	// An object that represents the specification of a gRPC gatewayRoute.
+	// +optional
+	GRPCRoute *GRPCGatewayRoute `json:"grpcRoute,omitempty"`
+	// An object that represents the specification of an HTTP gatewayRoute.
+	// +optional
+	HTTPRoute *HTTPGatewayRoute `json:"httpRoute,omitempty"`
+	// An object that represents the specification of an HTTP/2 gatewayRoute.
+	// +optional
+	HTTP2Route *HTTPGatewayRoute `json:"http2Route,omitempty"`
+	// A reference to k8s VirtualGateway CR that this GatewayRoute belongs to.
```
Comment (rajal-amzn) [inline apis/appmesh/v1beta2/gatewayroute_types.go:87]: nit: Remove "CR"

**Comment by karanvasnani** [comment]
Updated the doc links to point to "https://docs.aws.amazon.com/app-mesh/latest/userguide/virtual_gateways.html"

File: apis/appmesh/v1beta2/types.go [hunk 2]
```diff
@@ -189,6 +189,16 @@ type Listener struct {
...
 }
 
+// VirtualGatewayReference holds a reference to VirtualGateway.appmesh.k8s.aws
+type VirtualGatewayReference struct {
+	// Namespace is the namespace of VirtualGateway CR.
+	// If unspecified, defaults to the referencing object's namespace
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+	// Name is the name of VirtualGateway CR
+	Name string `json:"name"`
```
Comment (M00nF1sh) [inline apis/appmesh/v1beta2/types.go:199]: better to add a UID here for system populated reference, to make sure the real object referenced is same one.

**Pull request merged**